tests/drivers/counter/counter_basic_api/boards/nrf9160_pca10090.conf

```diff
@@ -0,0 +1,7 @@
+CONFIG_COUNTER_TIMER0=y
+CONFIG_COUNTER_TIMER0_PRESCALER=4
```

Comment:
Can we have the SoC code enable the various timers instead of having to add a board file?

Reply:
not sure what do you mean. The configuration is different for boards as they have different number of RTCs and TIMERs.

Reply:
How is the number of RTCs & TIMERs board specific?

Reply:
it's not board specific though every board here has different soc. Still i don't know what do you mean by `Soc code` in:
> Can we have the SoC code enable the various timers

I don't want to enable all timers by default in those soc. Only in that test.

Reply:
The issue I see, here, is that if someone wishes to have this test running for their nrf- board, they still need to add a board.conf. And we have ~20-25 nRF boards in the tree....

Reply:
On the other hand we don't want to enable all counter instances by default. On the other hand, what is the value in testing same driver on different boards with same soc?
However, I see you point and ideally it would be good to have option to add `.conf` file for soc that would be applied to any board with given soc (like `nrf9160.conf`). @galak, @ioannisg  Do we have option like that?

Reply:
We don't have something like that, no. At least not that I am aware of @nordic-krch

Reply:
@galak @ioannisg can we conclude here?
We have two options:
- enable all instances in soc (i'm against since it's a waste of ram and rom, some you cannot enable like TIMER0 and RTC0 which are used by Bluetooth)
- accept that each board that targets the test requires own board configuration

there is a third option to extend cmake to take conf files from `<app>\soc` folder. Then we would have specific option per soc which has value but i looked into cmake framework and it's not straightforward to do because contrary to `BOARD`, SoC name is taken from kconfig.

Reply:
@galak @ioannisg ping.

Reply:
I think the second option is decent enough to be accepted